Zend/zend_compile.h

```diff
@@ -382,6 +383,7 @@ typedef struct _zend_class_constant {
 	zval value; /* access flags are stored in reserved: zval.u2.access_flags */
 	zend_string *doc_comment;
 	HashTable *attributes;
+	zend_ast_attr const_flags;
```

Comment:
`zval value; /* access flags are stored in reserved: zval.u2.access_flags */` - I'm pretty sure this is where private|protected|public flags on class constants are already tracked.

If there is a reason why zval.u2.access_flags could not be reused for tracking ZEND_CLASS_CONST_IS_CASE, document it?

This may have gotten lost with your other comment about renaming from attr to const_flags

Reply:
> If there is a reason why zval.u2.access_flags could not be reused for tracking ZEND_CLASS_CONST_IS_CASE, document it?

Hm, just that it might be slightly surprising (especially given the naming). Also just noticed that we're using `zend_ast_attr` here, should probably just be `uint16_t`.

Reply:
```
	union {
		uint32_t     next;                 /* hash collision chain */
		uint32_t     cache_slot;           /* cache slot (for RECV_INIT) */
		uint32_t     opline_num;           /* opline number (for FAST_CALL) */
		uint32_t     lineno;               /* line number (for ast nodes) */
		uint32_t     num_args;             /* arguments number for EX(This) */
		uint32_t     fe_pos;               /* foreach position */
		uint32_t     fe_iter_idx;          /* foreach iterator index */
		uint32_t     access_flags;         /* class constant access flags */
		uint32_t     property_guard;       /* single property guard */
		uint32_t     constant_flags;       /* constant flags */
		uint32_t     extra;                /* not further specified */
	} u2;
```

Could change the comment to `/* class constant access flags and other flags */`

Or add `uint32_t class_constant_flags` to the union and change the Z_ACCESS_FLAGS macro to Z_CLASS_CONSTANT_FLAGS (keeping the old macro for third party pecl code)

Or just don't care about the name - it's already being used for things that aren't access flags and changing that is out of scope of this RFC, e.g. the macro `#define MARK_CONSTANT_VISITED(zv)   Z_ACCESS_FLAGS_P(zv) |= IS_CONSTANT_VISITED_MARK`

Reply:
Unfortunately this causes issues with differentiating cases and constants. The problem is that the flags will be copied when evaluating the constants, incorrectly resulting in them being flagged as cases. We could probably work around this by removing the flag when evaluating constants but I'm not sure that's worth it.

Reply:
Must've been a mistake because it's working now.